webserver/views/datasets.py

```diff
@@ -167,10 +171,23 @@ def evaluate(dataset_id):
         try:
             if form.filter_type.data == forms.DATASET_EVAL_NO_FILTER:
                 form.filter_type.data = None
+            # take default value
+            # if invalid non-numerical c and gamma
+            try:
+                c_value = [int(value) for value in form.c_value.data.split(',')]
+            except:
+                c_value = C
+            try:
+                gamma_value = [int(value) for value in form.gamma_value.data.split(',')]
+            except:
+                gamma_value = gamma
             db.dataset_eval.evaluate_dataset(
                 dataset_id=ds["id"],
                 normalize=form.normalize.data,
                 eval_location=form.evaluation_location.data,
+                c_value=c_value,
+                gamma_value=gamma_value,
+                preprocessing_values=form.preprocessing_values.data,
                 filter_type=form.filter_type.data,
             )
             flash.info("Dataset %s has been added into evaluation queue." % ds["id"])
```

Comment:
I think flashing a message here if the input values were bad would be a good idea too.